src/sync/mpsc/tip.rs

```diff
@@ -0,0 +1,135 @@
+use std::sync::{Arc, Mutex};
+
+use task::{self, Task};
+use {Sink, Stream, AsyncSink, Async, Poll, StartSend};
+
+/// Tip is very similar to unbounded channel but only stores last value sent
+///
+/// I.e. if you want to send some value between from producer to a consumer
+/// and if consumer is slow it should may skip old values, the tip is
+/// a structure for the task.
+
+/// The transmission end of a channel which is used to send values
+///
+/// If the receiver is not fast enough only the last value is preserved and
+/// other ones are discarded.
+#[derive(Debug)]
+pub struct TipSender<T> {
+    inner: Arc<Mutex<Inner<T>>>,
+}
+
+/// The receiving end of a channel which preserves only the last value
+#[derive(Debug)]
+pub struct TipReceiver<T> {
+    inner: Arc<Mutex<Inner<T>>>,
+}
+
+#[derive(Debug)]
+struct Inner<T> {
+    closed: bool,
+    value: Option<T>,
+    task: Option<Task>,
+}
+
+trait AssertKindsSender: Send + Sync + Clone {}
+impl AssertKindsSender for TipSender<u32> {}
+
+trait AssertKindsReceiver: Send + Sync {}
+impl AssertKindsReceiver for TipReceiver<u32> {}
+
+impl<T> TipSender<T> {
+    /// Sets the new new value of the stream and notifies the consumer if any
+    pub fn set(&self, value: T) {
+        // Do this step first so that the lock is dropped when
+        // `unpark` is called
+        let task = {
+            let mut inner = self.inner.lock().unwrap();
+            inner.value = Some(value);
+            inner.task.take()
+        };
+        if let Some(task) = task {
+            task.notify();
+        }
+    }
+}
+
+impl<T> Sink for TipSender<T> {
+    type SinkItem = T;
+    type SinkError = ();  // actually void
+    fn start_send(&mut self, item: T) -> StartSend<T, ()> {
+        self.set(item);
+        Ok(AsyncSink::Ready)
+    }
+    fn poll_complete(&mut self) -> Poll<(), Self::SinkError> {
+        Ok(Async::Ready(()))
+    }
+    fn close(&mut self) -> Poll<(), Self::SinkError> {
+        let task = {
+            let mut inner = self.inner.lock().unwrap();
+            inner.closed = true;
```

Comment:
I think this may not be quite right? This is only done when all senders have gone away, right?

I'd imagine that doing this in the destructor is fine, `close` is generally intended for shutdown logic like "tell the remote end that we're going away"